smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpUrlPathMiddleware.kt

```diff
@@ -50,20 +51,29 @@ class HttpUrlPathMiddleware(
                 // shape must be string, number, boolean, or timestamp
                 val targetShape = ctx.model.expectShape(binding.member.target)
                 val labelMemberName = ctx.symbolProvider.toMemberNames(binding.member).first.decapitalize()
-                val formattedLabel: String
-                if (targetShape.isTimestampShape) {
-                    val bindingIndex = HttpBindingIndex.of(ctx.model)
-                    val timestampFormat = bindingIndex.determineTimestampFormat(
-                        targetShape,
-                        HttpBinding.Location.LABEL,
-                        TimestampFormatTrait.Format.DATE_TIME
-                    )
-                    formattedLabel = ProtocolGenerator.getFormattedDateString(timestampFormat, labelMemberName)
-                } else if (targetShape.isStringShape) {
-                    val enumRawValueSuffix = targetShape.getTrait(EnumTrait::class.java).map { ".rawValue" }.orElse("")
-                    formattedLabel = "$labelMemberName$enumRawValueSuffix"
-                } else {
-                    formattedLabel = labelMemberName
+                val formattedLabel: String = when (targetShape.type) {
+                    ShapeType.TIMESTAMP -> {
+                        val bindingIndex = HttpBindingIndex.of(ctx.model)
+                        val timestampFormat = bindingIndex.determineTimestampFormat(
+                            binding.member,
+                            HttpBinding.Location.LABEL,
+                            TimestampFormatTrait.Format.DATE_TIME
+                        )
+                        ProtocolGenerator.getFormattedDateString(
+                            timestampFormat,
+                            labelMemberName,
+                            roundEpoch = true,
+                            urlEncode = true
+                        )
+                    }
+                    ShapeType.STRING -> {
+                        val percentEncoded = if (!it.isGreedyLabel) ".urlPercentEncoding()" else ""
+                        val enumRawValueSuffix =
+                            targetShape.getTrait(EnumTrait::class.java).map { ".rawValue" }.orElse("")
+                        "$labelMemberName$enumRawValueSuffix$percentEncoded"
+                    }
+                    ShapeType.FLOAT, ShapeType.DOUBLE -> "$labelMemberName.encoded()"
```

Comment:
i'm surprised that floats and doubles don't need percent encoding... just to clarify my understanding, we are basically saying a url path can have:
`/something/1.0/somethingElse`
?

Reply:
this is valid correct, verified per the protocol tests